6 create team page

Ram-Raghav-S opened the pull request:
The team page at the moment is made to work for large screens. It does have problems with scaling the text for medium and small devices - this will be fixed in a separate issue. 

Do let me know if any changes are required.

Ram-Raghav-S committed:
setup basic folder structure

Ram-Raghav-S committed:
Merge branch 'main' of https://github.com/WCSAorientation/OrientationWebsite2023

Ram-Raghav-S committed:
merged with remote main

Ram-Raghav-S committed:
refactored navbar to outside pages

Ram-Raghav-S committed:
Added /team to routes

Ram-Raghav-S committed:
added TeamTile component

Ram-Raghav-S committed:
imported all Roboto font weights

Ram-Raghav-S committed:
added temporary team images

Ram-Raghav-S committed:
added unique key prop to list widgets

Ram-Raghav-S committed:
Merge branch 'main' of https://github.com/WCSAorientation/OrientationWebsite2023

Ram-Raghav-S committed:
made team tile partially responsive

Ram-Raghav-S committed:
fixed font size

Ram-Raghav-S committed:
added additional demo image

Ram-Raghav-S committed:
added left/right side functionality

Ram-Raghav-S committed:
populated team page

Ram-Raghav-S committed:
populated team page with temporary data

Ram-Raghav-S committed:
added dynamic team info tile rendering

PierreLessard committed:
fix rounded bg, add img alt

PierreLessard reviewed (dismissed):
There was a rounding utility class being used on the team card div causing the corners to look a little cut off compared to the figma so I removed it and also added some img alts to get rid of the warnings. Otherwise, really nice! approved

Ram-Raghav-S committed:
fixed partially implemented dynamic rendering

Ram-Raghav-S committed:
changed rounding width

Ram-Raghav-S commented:
@PierreLessard I added the rounding in my original code because leaving it out caused an issue with containing certain images. The images in the original PR did not show this issue, but I have now added another image (the cat) to the list for which this issue shows up. You can try to remove the rounding from my code and test it out.

I have tried to add the rounding back with a different radius, but I agree that it still looks cut off compared to the figma. 

I think we will have to find another way of getting around this issue other than removing the rounding. Otherwise, it is also possible to preprocess images before adding them to the page.

Let me know what you think is best here.

PierreLessard committed:
team card corner fix

PierreLessard commented:
I fixed the corner issue by going into figma and creating a subtracted shape svg for that corner and laid it above the image. I had to add a few pixels below as well to prevent the image clipping below during graphical issues. @Ram-Raghav-S I want to get the ok from you that you think this method works before I approve it.

Ram-Raghav-S commented:
@PierreLessard It looks great on a large screen but it causes problems when we resize the screen. Would we be to do something that makes it looks good on small screens as well in the future? If yes, I think this is ok.

@takeachangs what do you think about this?

PierreLessard reviewed (approved):
Seongmin just let me know to worry about mobile responsiveness when we are done the rest. Also he said that most images will be cropped in a way that we dont have to worry about that corner.

takeachangs commented on frontend/src/components/TeamTile.js:
Good on desktop view, but resizing for mobile view is a little wonky. Don't forget to check in mobile view too for future edits!

takeachangs reviewed (approved):
Great that you created a separate data file!
